### ISSUE TITLE
get an error when using `*` in Static Site `http.alias`

### ISSUE THREAD

**Issue by YiCChi** [aws/copilot-cli#4990]
This is my manifest file.
```yml
name: client
type: Static Site
files:
  -  source: ./dist
     recursive: true

environment:
  test:
    http:
      alias: '*.test.mydomain.com'
```
And I'm using cdk to override `AWS::CloudFront::Distribution` to customize error response.

when running `copilot svc deploy` I got this.
```
rootPath/copilot/client/overrides/node_modules/yaml/dist/resolveSeq-d03cb037.js:1568
errors.push(new PlainValue.YAMLReferenceError(code, msg));
YAMLReferenceError: Aliased anchor not found: .test.mydomain.com
```

OS  Version: windows 10.
copilot Version: 1.28.0

**Comment by huanjani** [comment]
Hello, @YiCChi!

It looks like the `*` is being misinterpreted as a YAML alias and that's why it's complaining about no anchor. What happens if you use double quotes: `alias: "*.test.mydomain.com"`? 🤞🏼

**Comment by YiCChi** [comment]
Thank you, @huanjani !
I tried your suggestion but still got the same error.
I don't know if this error is due to the spec of YAML or this library `eemeli/yaml`.
If it's a library problem, I think maybe I can override `alias` using cdk.

**Comment by iamhopaul123** [comment]
Hello @YiCChi.

> And I'm using cdk to override AWS::CloudFront::Distribution to customize error response.

Would you mind to share your CDK overriding file. Or does the same error show up when you don't use CDK override?

**Comment by YiCChi** [comment]
Hello @iamhopaul123 .
Of course. Here is my `stack.ts`.

```typescript
import * as cdk from 'aws-cdk-lib';
import * as path from 'path';

interface TransformedStackProps extends cdk.stackProps {
  readonly appName: string;
  readonly envName: string;
}

export class TransformedStack extends cdk.Stack {
  public readonly template: cdk.cloudformation_include.CfnInclude;
  public readonly appName: string;
  public readonly envName: string;

  constructor(scope: cdk.App, id: string, props: TransformedStackProps) {
    super(scope, id, props);
    this.templage = new cdk.cloudformation_include.CfnInclude(this, 'Template', {
      templateFile: path.join('.build', 'in.yml'),
    });
    this.appName = props.appName;
    this.envName = props.envName;
    this.transformCloudFrontDistribution();
  }

  transformCloudFrontDistribution() {
    const cfd = this.template.getResource('CloudFrontDistribution') as cdk.aws_cloudfront.CfnDistribution;
    cfd.addPropertyOverride('DistributionConfig.CustomErrorResponses', [
      {
        ErrorCode: '403',
        ResponsePagePath: '/index.html',
        ResponseCode: '200',
        ErrorCachingMinTTL: '30',
      }
    ]);
    cfd.addPropertyOverride('DistributionConfig.DefaultRootObject', 'index.html');
  }
}
```

I haven't tried without overriding. Maybe I can try it today.

**Comment by iamhopaul123** [comment]
Hello @YiCChi. I was able to reproduce the error, and it went away when i don't use `*` in the alias. It seems to be a bug IMO and will have a fix out soon. <strike>In the mean time, would you mind to use the yaml patch doing the same thing? Here's the example `cfn.patches.yml` file:

```yaml
- op: add
  path: /Resources/CloudFrontDistribution/Properties/DistributionConfig/CustomErrorResponses
  value:
    - ErrorCode: 403
      ResponseCode: 200
      ResponsePagePath: /index.html
      ErrorCachingMinTTL: 30
```
</strike> Seems like YAML patch also has this issue.

And for `DefaultRootObject: index.html` I think we've already set that in the template Copilot generates.

**Comment by YiCChi** [comment]
> Hello @YiCChi. I was able to reproduce the error, and it went away when i don't use `*` in the alias. It seems to be a bug IMO and will have a fix out soon. <strike>In the mean time, would you mind to use the yaml patch doing the same thing? Here's the example `cfn.patches.yml` file:
> 
> 
> 
> ```yaml
> 
> - op: add
> 
>   path: /Resources/CloudFrontDistribution/Properties/DistributionConfig/CustomErrorResponses
> 
>   value:
> 
>     - ErrorCode: 403
> 
>       ResponseCode: 200
> 
>       ResponsePagePath: /index.html
> 
>       ErrorCachingMinTTL: 30
> 
> ```
> 
> </strike> Seems like YAML patch also has this issue.
> 
> 
> 
> And for `DefaultRootObject: index.html` I think we've already set that in the template Copilot generates.

thanks.I will update my stack.ts.

**Comment by huanjani** [comment]
This fix has been released in v1.29.0: https://github.com/aws/copilot-cli/releases/tag/v1.29.0. Thanks so much for bringing this to our attention!
